### PR TITLE
CASSANDRA-17544 Rename DisableFlag to EnableFlag on Guardrails

### PR DESCRIPTION
**Pull request by bbotella**
CASSANDRA-17544

### CODE REVIEW

File: src/java/org/apache/cassandra/db/guardrails/EnableFlag.java [hunk 1]
```diff
@@ -24,35 +24,35 @@
 import org.apache.cassandra.service.ClientState;
 
 /**
- * A guardrail that completely disables the use of a particular feature.
+ * A guardrail that enables the use of a particular feature.
  *
- * <p>Note that this guardrail only aborts operations (if the feature is disabled) so is only meant for
+ * <p>Note that this guardrail only enables operations (if the feature is enables) so is only meant for
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/db/guardrails/EnableFlag.java]: The original content `only aborts operations (if the feature is disabled)` is easier to understand.

File: src/java/org/apache/cassandra/db/guardrails/EnableFlag.java [hunk 2]
```diff
@@ -24,35 +24,35 @@
...
     /**
-     * Creates a new {@link DisableFlag} guardrail.
+     * Creates a new {@link EnableFlag} guardrail.
      *
      * @param name     the identifying name of the guardrail
-     * @param disabled a {@link ClientState}-based supplier of boolean indicating whether the feature guarded by this
-     *                 guardrail must be disabled.
+     * @param enabled a {@link ClientState}-based supplier of boolean indicating whether the feature guarded by this
+     *                 guardrail is enabled.
      * @param what     The feature that is guarded by this guardrail (for reporting in error messages),
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/db/guardrails/EnableFlag.java]: I know it is not part of the patch... it just feels a bit unclear to use `what` as a var name. Probably use `featureName`.

File: src/java/org/apache/cassandra/db/guardrails/EnableFlag.java [hunk 3]
```diff
@@ -24,35 +24,35 @@
...
-     * @param name     the identifying name of the guardrail
-     * @param disabled a {@link ClientState}-based supplier of boolean indicating whether the feature guarded by this
-     *                 guardrail must be disabled.
-     * @param what     The feature that is guarded by this guardrail (for reporting in error messages),
-     *                 {@link DisableFlag#ensureEnabled(String, ClientState)} can specify a different {@code what}.
+     * @param name The identifying name of the guardrail
+     * @param enabled a {@link ClientState}-based supplier of boolean indicating whether the feature guarded by this
+     *                guardrail is enabled.
+     * @param featureName The feature that is guarded by this guardrail (for reporting in error messages),
+     *                    {@link EnableFlag#ensureEnabled(String, ClientState)} can specify a different {@code featureName}.
```
Comment (adelapena) [inline src/java/org/apache/cassandra/db/guardrails/EnableFlag.java]: Nit: This is changing the way the JavaDoc parameters are aligned. No clue if we have any official guideline on how these should be aligned and one can find different approaches across the codebase. I'd keep the original format in the absence of a clear guideline.

File: src/java/org/apache/cassandra/db/guardrails/Guardrails.java [hunk 4]
```diff
@@ -100,9 +100,9 @@ public final class Guardrails implements GuardrailsMBean
     /**
      * Guardrail disabling user's ability to create secondary indexes
      */
-    public static final DisableFlag createSecondaryIndexesEnabled =
-    new DisableFlag("secondary_indexes",
-                    state -> !CONFIG_PROVIDER.getOrCreate(state).getSecondaryIndexesEnabled(),
+    public static final EnableFlag createSecondaryIndexesEnabled =
+    new EnableFlag("secondary_indexes",
+                    state -> CONFIG_PROVIDER.getOrCreate(state).getSecondaryIndexesEnabled(),
```
Comment (adelapena) [inline src/java/org/apache/cassandra/db/guardrails/Guardrails.java]: Nit: The parameters of all the six `EnableFlag` constructors are now misaligned, they should be:
```java
    public static final EnableFlag createSecondaryIndexesEnabled =
    new EnableFlag("secondary_indexes",
                   state -> CONFIG_PROVIDER.getOrCreate(state).getSecondaryIndexesEnabled(),
                   "User creation of secondary indexes");
```

File: src/java/org/apache/cassandra/db/guardrails/EnableFlag.java [hunk 5]
```diff
@@ -62,25 +62,25 @@ public DisableFlag(String name, Predicate<ClientState> disabled, String what)
...
 
     /**
-     * Aborts the operation if this guardrail is disabled.
+     * Aborts the operation if this guardrail is not enabled.
      *
      * <p>This must be called when the feature guarded by this guardrail is used to ensure such use is in fact
      * allowed.
      *
-     * @param what  The feature that is guarded by this guardrail (for reporting in error messages).
+     * @param featureName  The feature that is guarded by this guardrail (for reporting in error messages).
```
Comment (adelapena) [inline src/java/org/apache/cassandra/db/guardrails/EnableFlag.java]:
```suggestion
     * @param featureName The feature that is guarded by this guardrail (for reporting in error messages).
```

**Review by adelapena — COMMENTED** [review]
Looks good to me, I have left a couple of nits about formatting. We'll need a rebase to integrate a couple of additional flags that have been added in the meantime, I can run CI once we have those.

File: test/unit/org/apache/cassandra/db/guardrails/GuardrailsTest.java [hunk 6]
```diff
@@ -250,26 +250,25 @@ public void testMinThresholdUsers() throws Throwable
         assertValid(() -> guard.guard(9, "z", false, superClientState));
     }
 
-    @Test
-    public void testDisableFlag() throws Throwable
+    public void testEnableFlag() throws Throwable
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/db/guardrails/GuardrailsTest.java:253]: This is missing the `@Test` tag, so it won't be run.

Comment (yifan-c) [reply test/unit/org/apache/cassandra/db/guardrails/GuardrailsTest.java:253]: Nice catch!

**Review by yifan-c — APPROVED** [review]

File: test/unit/org/apache/cassandra/db/guardrails/GuardrailsTest.java (hunk 6 above)
Comment (bbotella) [reply test/unit/org/apache/cassandra/db/guardrails/GuardrailsTest.java:253]: Yeah! Thanks!

**Pull request closed without merging**